### PR TITLE
Add more Golang tests to filestream input

### PR DESCRIPTION
**Pull request by kvch**
## What does this PR do?

Increase the test coverage of `filestream` input.

# Related issues

Requires #22448

### CODE REVIEW

File: filebeat/input/filestream/prospector_test.go [hunk 1]
```diff
@@ -156,14 +255,98 @@ func TestProspectorDeletedFile(t *testing.T) {
...
+
+			has := testStore.has("path::/old/path/to/file")
+			if test.trackRename {
+				assert.True(t, has)
+			} else {
+				assert.False(t, has)
+			}
+
+			assert.ElementsMatch(t, test.expectedEncounteredNames, hg.encounteredNames)
+			assert.ElementsMatch(t, test.expectedStoppedNames, hg.stoppedNames)
```
Comment (urso) [inline filebeat/input/filestream/prospector_test.go:326]: These tests seem to aim for checking behavior and not state. The `hg` is collecting the output and the mockFileWatcher is the input. Given that input events have some order, I would expect that the order in the output is important as well. How about storing the `harvesterGroup` calls as an event stream and verifying it?

e.g.

```
type harvesterEvent interface{ String() string }
type harvesterStart string
type harvesterStop string
type harvesterGroupStop struct {}



type testHarvesterGroup struct {
  events []harvesterEvent
}

...

want := []harvesterEvent{
  harvesterStart("..."),
  harvesterStop("..."),
}
```

File: filebeat/input/filestream/fswatch_test.go [hunk 2]
```diff
@@ -93,12 +104,14 @@ func TestFileScanner(t *testing.T) {
...
 
 func setupFilesForScannerTest(t *testing.T) {
-	err := os.MkdirAll(directoryPath, 0750)
-	if err != nil {
-		t.Fatal(t)
+	for _, dir := range []string{unharvestableDir, includedDir} {
+		err := os.MkdirAll(dir, 0750)
+		if err != nil {
+			t.Fatal(t)
+		}
```
[the remark this answers is not in the document]
Comment (kvch) [reply filebeat/input/filestream/fswatch_test.go:111]: I will move it to a TempDir.

File: filebeat/input/filestream/identifier_test.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+	loginp "github.com/elastic/beats/v7/filebeat/input/filestream/internal/input-logfile"
+	"github.com/elastic/beats/v7/libbeat/common"
+)
+
+type testFileIdentifierConfig struct {
+	Identifier *common.ConfigNamespace `config:"identifier"`
+}
+
+func TestNewFileIdentifier(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (kvch) [reply filebeat/input/filestream/identifier_test.go:34]: Yes, now.

File: filebeat/input/filestream/identifier_test_non_windows.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+		_, err = markerFile.Write([]byte(markerContents))
+		if err != nil {
+			t.Fatalf("cannot write marker contents to file: %v", err)
+		}
+		markerFile.Sync()
+
+		c := common.MustNewConfigFrom(map[string]interface{}{
+			"identifier": map[string]interface{}{
+				"inode_marker": map[string]interface{}{
+					"path": markerFile.Name(),
```
Comment (urso) [inline filebeat/input/filestream/identifier_test_non_windows.go:53]: btw. you can save some lines here by writing `"identifier.inode_marker.path": markerFile.Name()`

**Review by urso — APPROVED** [review]

**Pull request merged**